bson/bson_test.go

```diff
@@ -202,3 +202,34 @@ func TestExtJSONEscapeKey(t *testing.T) {
 		t.Errorf("Unmarshaled documents do not match. got %v; want %v", got, doc)
 	}
 }
+
+type BSONDocumentArray struct {
```

Comment:
Let's move the type definitions for `BSONDocumentArray` and `BSONArray` into the test itself so it's clear that they're only used in the scope of the one test.

Reply:
Sure!